src/transformers/models/oneformer/modeling_oneformer.py

```diff
@@ -0,0 +1,3256 @@
+# coding=utf-8
+# Copyright 2022 SHI Labs and The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" PyTorch OneFormer model."""
+import copy
+import math
+import warnings
+from dataclasses import dataclass
+from numbers import Number
+from typing import Dict, List, Optional, Tuple
+
+import numpy as np
+import torch
+import torch.nn.functional as F
+from torch import Tensor, nn
+from torch.cuda.amp import autocast
+
+from transformers import AutoBackbone, MaskFormerSwinConfig
```

Comment:
MaskFormerSwinConfig should not be used in this PR, either we can first add the checkpoints that use DiNAT as backbone and later on add Swin, or wait for #20648 to be merged such that SwinBackbone can be leveraged straightaway